File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode removeNthFromEnd(ListNode head, int n) {

        if(head==null || head.next==null) return null;

        ListNode temp = new ListNode();
        temp.next = head;
        ListNode first = temp, second = temp;

        while(n > 0){
            second = second.next;
            n--;
        }
        while(second.next !=null){
            first=first.next;
            second = second.next;
        }
        first.next = first.next.next;
        return temp.next;
    }
}

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode(object):
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution(object):
    def removeNthFromEnd(self, head, n):
        """
        :type head: ListNode
        :type n: int
        :rtype: ListNode
        """
        d= ListNode(0)
        d.next= head
        slow = d
        fast = d
        i=0
        while i<n:
            fast = fast.next
            i+=1
        while fast and fast.next:
            fast = fast.next
            slow = slow.next
        slow.next= slow.next.next
        return d.next
                
        
        